### PR TITLE
Add TS examples and section on changedProperties.

### LINKED ISSUE #774: Lifecycle code examples should be switchable TS/JS
Currently they're all in JavaScript, so the user has to guess at the correct typing for `changedProperties`, for example.

Which should be `PropertyValues<this>` I think...

### PR DESCRIPTION
**Pull request by arthurevans**
Fixes #774.

### CODE REVIEW

File: packages/lit-dev-content/site/docs/components/lifecycle.md [hunk 1]
```diff
@@ -147,6 +147,22 @@ In more detail, it looks like this:
 
 <img class="centered-image" src="/images/docs/components/update-4.jpg">
 
+### The changedProperties map {#changed-properties}
+
+Many reactive update methods receive a `Map` of changed properties. The `Map` keys are the property names and its values are the previous property values. Changing a property *during* the update  (up to and including the `render()` method) updates the `changedProperties` map, but does not kick off a new update. Changing a property _after_ `render()` (for example, in the `updated()` method) triggers a new update cycle, and the changed property is added to a new `changedProperties` map to be used for he next cycle.
```
Comment (justinfagnani) [inline packages/lit-dev-content/site/docs/components/lifecycle.md:152]: Maybe move the mention of updating properties during an update to after the types section? It's kind of an edge case.

[the remark this answers is not in the document]
Comment (arthurevans) [reply packages/lit-dev-content/site/docs/components/lifecycle.md:152]: Yeah. If I was doing that, I'd just do `Map<string>` and be done with it... But seriously, what is the use case for doing `requestUpdate(name, oldValue)` for a _non_ reactive property? It seems like you'd have to do a bunch of extra work (like keeping track of previous values) for dubious benefit.

[the remark this answers is not in the document]
Comment (arthurevans) [reply packages/lit-dev-content/site/docs/components/lifecycle.md:152]: Sounds good.

Comment (arthurevans) [reply packages/lit-dev-content/site/docs/components/lifecycle.md:152]: Hmm... I was afraid if I put it after the types, folks who weren't using typescript would never get there. Maybe I'll make these two little H4 subsections--TypeScript types and Changing properties during an update

[the remark this answers is not in the document]
Comment (arthurevans) [reply packages/lit-dev-content/site/docs/components/lifecycle.md:152]: After talking offline, I think the conclusion here was that this was footgun-y and probably not something we want to document.

Comment (arthurevans) [reply packages/lit-dev-content/site/docs/components/lifecycle.md]: Did that, I think it works pretty well.

**Review by justinfagnani — APPROVED** [review]

**Pull request merged**